BlazarService/src/main/java/com/hubspot/blazar/guice/BlazarSlackModule.java

```diff
@@ -0,0 +1,77 @@
+package com.hubspot.blazar.guice;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.multibindings.Multibinder;
+import com.google.inject.name.Names;
+import com.hubspot.blazar.base.visitor.RepositoryBuildVisitor;
+import com.hubspot.blazar.config.BlazarConfiguration;
+import com.hubspot.blazar.config.BlazarSlackConfiguration;
+import com.hubspot.blazar.listener.SlackDmNotificationVisitor;
+import com.hubspot.blazar.listener.SlackRoomNotificationVisitor;
+import com.hubspot.blazar.resources.SlackResource;
+import com.hubspot.blazar.resources.UserFeedbackResource;
+import com.hubspot.blazar.util.SlackClientWrapper;
+import com.hubspot.blazar.util.SlackMessageBuildingUtils;
+import com.ullink.slack.simpleslackapi.SlackSession;
+import com.ullink.slack.simpleslackapi.impl.SlackSessionFactory;
+
+/**
+ * This module handles the configuration of all slack-related features inside of Blazar.
+ * This module does not bind any slack related resources / clients / visitors etc. if slack is not configured.
+ */
+public class BlazarSlackModule implements Module {
+
+  public static final String SLACK_FEEDBACK_ROOM = "SLACK_FEDBACK_ROOM";
+  private static final Logger LOG = LoggerFactory.getLogger(BlazarSlackModule.class);
+  private final Optional<BlazarSlackConfiguration> slackConfiguration;
+  private final Multibinder<RepositoryBuildVisitor> repositoryBuildVisitorMultibinder;
+
+  public BlazarSlackModule(BlazarConfiguration configuration, Multibinder<RepositoryBuildVisitor> repositoryBuildVisitorMultibinder) {
+    this.slackConfiguration = configuration.getSlackConfiguration();
+    this.repositoryBuildVisitorMultibinder = repositoryBuildVisitorMultibinder;
+  }
+
+  @Override
+  public void configure(Binder binder) {
+
+    if (!slackConfiguration.isPresent()) {
+      LOG.info("Slack is not configured, not binding slack related resources or slack build notification visitors");
+      return;
+    }
+
+    binder.bind(SlackSession.class).toInstance(makeNewSlackSession());
```

Comment:
Why don't you create a provider method instead?
Don't you want the session to be a singleton? i.e. keep a single slack session along the app lifecycle? Do you need to get a new session instance each time you inject the session class?